plugins/inputs/nfsclient/README.md

```diff
@@ -0,0 +1,136 @@
+
+- nfs_bytes
+    - tags:
+        - mountpoint - The local mountpoint, for instance: "/var/www"
+        - serverexport - The full server export, for instance: "nfsserver.example.org:/export"
+    - fields:
+        - normalreadbytes - (int, bytes) - Bytes read from the server via `read()`
+        - normalwritebytes - (int, bytes) - Bytes written to the server via `write()`
+        - directreadbytes - (int, bytes) - Bytes read with O_DIRECT set
+        - directwritebytes - (int, bytes) -Bytes written with O_DIRECT set
+        - serverreadbytes - (int, bytes) - Bytes read via NFS READ (via `mmap()`)
+        - serverwritebytes - (int, bytes) - Bytes written via NFS WRITE (via `mmap()`)
+        - readpages - (int, count) - Number of pages read
+        - writepages - (int, count) - Number of pages written
+
+- nfs_events - Per-event metrics
+    - tags:
+        - mountpoint - The local mountpoint, for instance: "/var/www"
+        - serverexport - The full server export, for instance: "nfsserver.example.org:/export"
+    - fields:
+        - inoderevalidates - (int, count) - How many times cached inode attributes have to be re-validated from the server.
+        - dentryrevalidates - (int, count) - How many times cached dentry nodes have to be re-validated.
+        - datainvalidates - (int, count) - How many times an inode had its cached data thrown out.
+        - attrinvalidates - (int, count) - How many times an inode has had cached inode attributes invalidated.
+        - vfsopen - (int, count) - How many times files or directories have been `open()`'d.
+        - vfslookup - (int, count) - How many name lookups in directories there have been.
+        - vfsaccess - (int, count) - Number of calls to `access()`. (formerly called "vfspermission")
+
+        - vfsupdatepage - (int, count) - Count of updates (and potential writes) to pages.
+        - vfsreadpage - (int, count) - Number of pages read.
+        - vfsreadpages - (int, count) - Count of how many times a _group_ of pages was read (possibly via `mmap()`?).
+        - vfswritepage - (int, count) - Number of pages written.
+        - vfswritepages - (int, count) - Count of how many times a _group_ of pages was written (possibly via `mmap()`?)
+        - vfsgetdents - (int, count) - Count of directory entry reads with getdents(). These reads can be served from cache and don't necessarily imply actual NFS requests. (formerly called "vfsreaddir")
+        - vfssetattr - (int, count) - How many times we've set attributes on inodes.
+        - vfsflush - (int, count) - Count of times pending writes have been forcibly flushed to the server.
+        - vfsfsync - (int, count) - Count of calls to `fsync()` on directories and files.
+        - vfslock - (int, count) - Number of times a lock was attempted on a file (regardless of success or not).
+        - vfsrelease - (int, count) - Number of calls to `close()`.
+        - congestionwait - (int, count) - Unused.
```

Comment:
What does `unused` mean? If this field is not output, please remove. Alternatively, you could convert the whole fields thing to a table where the first column is the field-name, the you have two additional columns where you put a cross if the field is collected in "basic stats" and "fullstats", respectively. But just a suggestion, no must-have.

Reply:
The "unused" comment should be "unused within the kernel", and I'll amend the description.
I include it here because it is included in the mountstats file, and silently dropping/changing a field name from the canonical name in the source/specification has already bitten me once in this module.    

The "unused" comment comes from here:  https://utcc.utoronto.ca/~cks/space/blog/linux/NFSMountstatsBytesEvents  
Based on my reading of the kernel source, it exists as `NFSIOS_CONGESTIONWAIT,` in `include/linux/nfs_iostat.h`.  However, not accounting for it means that the other metrics that follow it, which _are_ used, have a different index in the tables. 

As for general formatting, I tried to keep the same style as the rest of the input plugins, not just this one.

Reply:
See a79c679a95577d9937cee2fc28be5584264bf385

Reply:
Please mark as resolved.